### ISSUE TITLE
Documentation for pagination with nested joins

### ISSUE THREAD

**Issue by nkezhaya** [mojotech/scrivener_ecto#75]
I just spent the past few minutes reading through the various issues/PRs on this: #1, #44, #45, etc., and I'm still unsure of the recommended approach for paginating a query with nested join(s).

In SQL, the approach is clear:

```sql
SELECT * FROM (SELECT * FROM posts LIMIT 10 OFFSET 20)
  LEFT JOIN comments ...
```

I can't find any documentation showing how this is to be done with Scrivener. I'm happy to submit a PR if necessary, but what's the recommended approach for this? Should this just be done with multiple queries?

**Comment by astjohn** [comment]
@whitepaperclip - As far as we can tell, the original issue was never fixed by @drewolson. 

[Here we have created a failing test once more as an example](https://github.com/Sitata/scrivener_ecto/commit/c6fc780b9521c9deccfe35a73d9594837ffdaf02)

In our test, we create 6 Posts, each with 5 comments. 

```
page =
        Post
        |> Post.published
        |> join(:inner, [p], c in assoc(p, :comments))
        |> preload([p, c], [comments: c])
        |> Scrivener.Ecto.Repo.paginate
```

In this example, `page.total_entries` ends up being 30 instead of 6. This is directly related to the fact that the preloads are not being treated properly by the current implementation and so entries are being returned for each comment instead of each post with nested comments.

We're going to once more create a fix on the same branch because #44 and #45 were never really addressed properly.

**Comment by nkezhaya** [comment]
@astjohn Thanks! Let me know if you need any help on this, because I definitely have bandwidth for the next couple weeks.

Right now I'm basically calling `Repo.preload` in a ton of functions because of the issue with `total_entries`, and the performance isn't ideal.

**Comment by astjohn** [comment]
@whitepaperclip - I have put back the original fix from #44, #45 so feel free to give it a spin as a temporary fix until we can have some discussion on perhaps a better approach.

https://github.com/drewolson/scrivener_ecto/compare/master...Sitata:fix_joins_pagination_2
